src/components/DiffView/index.tsx

```diff
@@ -0,0 +1,28 @@
+/// <reference path="index.d.ts"/>
+
+import * as React from 'react';
+import { Diff, DiffInfo, DiffProps, parseDiff } from 'react-diff-view';
+
+type Props = {
+  diff: string;
+  viewType: DiffProps['viewType'];
+};
+
+class DiffView extends React.Component<Props> {
+  static defaultProps = {
+    viewType: 'unified',
+  };
+
+  render = () => {
+    const { diff, viewType } = this.props;
+    const files = parseDiff(diff);
+
+    const renderFile = ({ hunks, type }: DiffInfo, index: number) => (
```

Comment:
TS can figure out `DiffInfo` if you pass it inline:

```diff
diff --git a/src/components/DiffView/index.tsx b/src/components/DiffView/index.tsx
index 88d020a..ba72c6c 100644
--- a/src/components/DiffView/index.tsx
+++ b/src/components/DiffView/index.tsx
@@ -17,11 +17,9 @@ class DiffView extends React.Component<Props> {
     const { diff, viewType } = this.props;
     const files = parseDiff(diff);
 
-    const renderFile = ({ hunks, type }: DiffInfo, index: number) => (
+    return <div>{files.map(({ hunks, type }, index: number) => (
       <Diff diffType={type} hunks={hunks} key={index} viewType={viewType} />
-    );
-
-    return <div>{files.map(renderFile)}</div>;
+    ))}</div>;
   };
 }
 
```

Reply:
Yep, that works.